lib/kredis/types/list.rb

```diff
@@ -10,14 +10,17 @@ def elements
 
   def remove(*elements, pipeline: nil)
     types_to_strings(elements, typed).each { |element| (pipeline || proxy).lrem 0, element }
+    (pipeline || proxy).expire expires_in.to_i if expires_in
```

Comment:
You can consider extracting a `refresh_expiration` method to reuse in the three invocations:

```ruby
def prepend(*elements, pipeline: nil)
  (pipeline || proxy).lpush types_to_strings(elements, typed) if elements.flatten.any?
  refresh_expiration(pipeline)
end

# ...

def refresh_expiration(pipeline)
  (pipeline || proxy).expire expires_in.to_i if expires_in
end
```